ocaml/testsuite/tests/typing-layouts/annots.ml

```diff
@@ -166,28 +276,73 @@ Error: This type string should be an instance of type ('a : immediate)
          of the definition of t2_imm at line 1, characters 0-28.
 |}]
 
-let f : 'a t2_imm -> 'a t2_imm = fun x -> x
+type t = string t2_global
+;;
+[%%expect {|
+Line 1, characters 9-15:
+1 | type t = string t2_global
+             ^^^^^^
+Error: This type string should be an instance of type ('a : value mod global)
+       The layout of string is value, because
```

Comment:
I also think we're going to have to switch these messages from saying "layout" to "kind". Actually, it's probably best to report layout if that's the interesting bit. But if the layouts are OK, we should print the kinds.

Reply:
Yes, I definitely agree that these errors are terrible. My thought was to leave the message as-is if the layouts disagree, but in other cases, say that the kinds disagree and give the full kind.

That being said, I intentionally didn't do that as part of this PR to keep it smaller. But it would make sense to me if an improved error message is a prerequisite to merging.

Reply:
I figured as much, about your intention. The problem is that merging this will, I think, open the possibility of these bad error messages. One way forward is to leave this PR as-is, with bad error messages, and then make a stacked PR, reviewed separately, that's just about the error messages. We'd merge the stacked PR into this one, and then merge into `main`. Does that sound like it would work?

Reply:
Yep, that sounds reasonable. We may have to lose our review history on this PR though as a consequence. Since the incoming branch is from my fork, I can't open a PR against it within this repo. Unless GitHub has a feature to do this that I'm not aware of, I'll have to close this PR, create a branch within this repo, and open a PR on that. The alternative would be to have a PR within my fork, but we'd lose CI and imo it'd be bad because then PRs wouldn't all be in one place.

I should probably just start making branches on this repo instead of on my fork to avoid these situations in the future...

Reply:
Done in #2730